### ISSUE TITLE
--dbt add snowflake SSO support

### ISSUE THREAD

**Issue by dlawin** [datafold/data-diff#413]
**Is your feature request related to a problem? Please describe.**
I am not able to use the `--dbt` functionality with snowflake sso
![image](https://user-images.githubusercontent.com/11282254/221244582-ea10195c-2269-4d75-8f25-69909786b90f.png)

**Describe the solution you'd like**
Support the connection method described here:  
https://docs.getdbt.com/reference/warehouse-setups/snowflake-setup#sso-authentication

**Describe alternatives you've considered**
The only option is to use password authentication, which may conflict with security policies

**Comment by lucasasmith** [comment]
+1 on this. Without SSO, I won't be able to use the package.

**Comment by dlawin** [comment]
> +1 on this. Without SSO, I won't be able to use the package.

ty Lucas this helps us prioritize, if anyone is lurking please feel free to comment and/or drop an emoji

**Comment by dlawin** [comment]
> +1 on this. Without SSO, I won't be able to use the package.

@lucasasmith would you be interested in testing this? I have a PR open but I'm not easily able to test it

If so you can install my branch like so `pip install git+https://github.com/dlawin/data-diff.git@issue_413 data-diff --ignore-installed`

To go back to the live version: `pip install data-diff --ignore-installed`

**Comment by lucasasmith** [comment]
> > +1 on this. Without SSO, I won't be able to use the package.
> 
> @lucasasmith would you be interested in testing this? I have a PR open but I'm not easily able to test it
> 
> If so you can install my branch like so `pip install git+https://github.com/dlawin/data-diff.git@issue_413 data-diff --ignore-installed`
> 
> To go back to the live version: `pip install data-diff --ignore-installed`

Yes. I should have time tomorrow and will get you feedback.

**Comment by lucasasmith** [comment]
> > +1 on this. Without SSO, I won't be able to use the package.
> 
> @lucasasmith would you be interested in testing this? I have a PR open but I'm not easily able to test it
> 
> If so you can install my branch like so `pip install git+https://github.com/dlawin/data-diff.git@issue_413 data-diff --ignore-installed`
> 
> To go back to the live version: `pip install data-diff --ignore-installed`

@dlawin It looks like SSO auth is working as I'm not seeing any auth errors. I'm trying to figure out why why my setup is using `db.schema_schema.table` for my prod table but I think that's a fully separate issue. Great work!

**Comment by dlawin** [comment]
> @dlawin It looks like SSO auth is working as I'm not seeing any auth errors. 
Awesome thanks for testing this!

> I'm trying to figure out why why my setup is using `db.schema_schema.table` for my prod table but I think that's a fully separate issue.

From what you described it's likely this issue: https://github.com/datafold/data-diff/issues/453
tldr; set this var:
```
vars:
  data_diff:
    custom_schemas: False
```
